### ISSUE TITLE
2.3.1 RC1 -- no such file or directory

### ISSUE THREAD

**Issue by thezoggy** [sabnzbd/sabnzbd#1054]
2.3.1RC1 [00436df] user just shared in irc,

```
2017-10-15 05:44:04,432::INFO::[database:239] Removing job SABnzbd_nzo_G_XqEi from history
....
2017-10-15 05:45:11,105::ERROR::[__init__:897] Saving /Downloads/incomplete/Star.Trek.The.Next.Generation.S04E11.Datas.Day.1080p.BluRay.x264/__ADMIN__/SABnzbd_nzo_G_XqEi failed
2017-10-15 05:45:11,105::INFO::[__init__:898] Traceback: 
Traceback (most recent call last):
  File "/usr/share/sabnzbdplus/sabnzbd/__init__.py", line 883, in save_data
    with open(path, 'wb') as data_file:
IOError: [Errno 2] No such file or directory: u'/Downloads/incomplete/Star.Trek.The.Next.Generation.S04E11.Datas.Day.1080p.BluRay.x264/__ADMIN__/SABnzbd_nzo_G_XqEi'
```

wondering if its:
https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/nzbqueue.py#L197

we should log the calling function so we can see what is calling `remove_history`.. or add log lines on entry to certain functions to help track stuff down..  cause theres x many ways remove_history gets ran.. and I dont know of which is the culprit that causes this scenario.

**Comment by Safihre** [comment]
Hmm that delete could have come from anywhere indeed.
I have added now to this call and also to all the saving/loading calls [this excellent function](https://gist.github.com/techtonik/2151727) that prints the caller name to the log.
It does make every call ~2-4ms slower.

Closing this for now, but re-open when we know the source of the problem. Could also be some race-issue in purging of history. I vaguely remember something like that happening to me a while back, purging all history would also break some jobs in queue.
